launcher/ui/dialogs/ModDownloadDialog.cpp

```diff
@@ -83,16 +115,38 @@ QList<BasePage *> ModDownloadDialog::getPages()
     };
 }
 
-void ModDownloadDialog::setSuggestedMod(const QString& name, ModDownloadTask* task)
+void ModDownloadDialog::addSelectedMod(const QString& name, ModDownloadTask* task)
+{
+    if(modTask.contains(name))
+        delete modTask.find(name).value();
+
+    if(task)
+        modTask.insert(name, task);
+    else
+        modTask.remove(name);
```

Comment:
I don't get the logic behind removing here. `task` is never (and probably shouldn't be) `nullptr`?

Reply:
You're right, before my changes the way to deselect the task was to pass `nullptr` to this function, but since there's now another function to handle this removal, this check is not needed anymore. I'll remove it!